kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala

```diff
@@ -71,23 +87,38 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
   def batchInfo(@PathParam("batchId") batchId: String): Batch = {
     try {
       val sessionHandle = sessionManager.getBatchSessionHandle(batchId, REST_BATCH_PROTOCOL)
-      val session = sessionManager.getSession(sessionHandle).asInstanceOf[KyuubiBatchSessionImpl]
-      buildBatch(session)
+      buildBatch(sessionHandle)
     } catch {
       case NonFatal(e) =>
         error(s"Invalid batchId: $batchId", e)
         throw new NotFoundException(s"Invalid batchId: $batchId")
     }
   }
 
-  private def buildBatch(session: KyuubiBatchSessionImpl): Batch = {
-    val batchOp = session.batchJobSubmissionOp
-    Batch(
-      batchOp.batchId,
-      batchOp.batchType,
-      batchOp.currentApplicationState.getOrElse(Map.empty),
-      fe.connectionUrl,
-      batchOp.getStatus.state.toString)
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON)),
+    description = "get operation log")
+  @GET
+  @Path("{batchId}/log")
+  def getBatchLog(
+      @PathParam("batchId") batchId: String,
+      @QueryParam("maxRows") maxRows: Int): OperationLog = {
```

Comment:
for restful api, it is stateless.

So I think it is better that we can specify the from offset when fetching logs.
If the from offset is not specified, we can always forward.

Reply:
cc @yaooqinn

Reply:
Looks fine to me, we can do this in another pr. The rest non-batch code path can also support this feature.

Reply:
https://github.com/apache/incubator-kyuubi/discussions/2304

Reply:
@zhaomin1423 thank you for the reminder !